src/FieldDefinition.hack

```diff
@@ -34,4 +35,16 @@ final class FieldDefinition<TParent, TRet, TResolved> implements IFieldDefinitio
 
         return await $this->type->resolveAsync($value, $field, $vars);
     }
+
+    public function getName(): string {
+        return $this->name;
+    }
+
+    public function getIntrospectionType(): Introspection\__Type {
+        if ($this->type is Introspection\INullableType) {
+            return $this->type->getInnerType();
+        }
+
+        return new Introspection\NonNullable($this->type as Introspection\__Type);
```

Comment:
Do we need some kind of recursion here if we're dealing with nested lists? e.g. `IntInputType::nonNullable()->nullableListOf()` needs removing the Nullable wrapper from the list and adding a NonNullable wrapper around the inner int type